pkg/dispers/target.go

```diff
@@ -0,0 +1,146 @@
+package enclave
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/cozy/cozy-stack/pkg/dispers/dispers"
+)
+
+func buildQuery(instance dispers.Instance, localQuery dispers.LocalQuery) dispers.Query {
+	// TODO : encrypt outputs
+	query := dispers.Query{
+		Domain:      instance.Domain,
+		LocalQuery:  localQuery,
+		TokenBearer: instance.Token.TokenBearer,
+	}
+
+	return query
+}
+
+func removeIndex(s *[]dispers.Instance, index int) []dispers.Instance {
+	return append((*s)[:index], (*s)[index+1:]...)
+}
+
+// cleanTargetsList clean Targets list inplace
+func cleanTargetsList(list *[]dispers.Instance) error {
+
+	i := 0
+	for i < len(*list) {
+		if isInstInvalid(&(*list)[i]) {
+			// Unvalid item
+			*list = removeIndex(list, i)
+			i--
+		} else {
+			// Valid item
+			// Check for duplicates
+			j := i + 1
+			iHasBeenDeleted := false
+			for !iHasBeenDeleted && j < len(*list) {
+				if (*list)[i].Domain == (*list)[j].Domain {
+					if (*list)[i].SubscriptionDate.After((*list)[j].SubscriptionDate) {
+						// Item i is kept
+						*list = removeIndex(list, j)
+						j--
+					} else {
+						// Item j is kept
+						*list = removeIndex(list, i)
+						iHasBeenDeleted = true
+					}
+				}
+				j++
+			}
+		}
+		i++
+	}
+	// Check missing data
+	return nil
+}
+
+func decryptInputT(in *dispers.InputT) error {
+	return nil
+}
+
+func retrieveData(in *dispers.InputT, queries *[]dispers.Query) ([]map[string]interface{}, error) {
+
+	var data []map[string]interface{}
+	var rowsData []map[string]interface{}
+
+	for _, query := range *queries {
+
+		url := strings.Join([]string{"http:/", query.Domain, "data/_find"}, "/")
```

Comment:
a `/` is missing after http

Reply:
It's kind of confusing to add the `/` after the 3 elements. Maybe use this instead 
```
url := &url.URL{
  Scheme:   "http",
  Host:     query.Domain,
  Path:     "data/_find/,
}
```

Reply:
Why the pointer ? I'm fine to put a pointer, I did it but why ? What does it change to have url as a pointer instead of having it as a url.URL object ?